staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -527,6 +527,16 @@ func (cnc *CloudNodeController) getProviderID(ctx context.Context, node *v1.Node
 		return node.Spec.ProviderID, nil
 	}
 
+	if instancesV2, ok := cnc.cloud.InstancesV2(); ok {
+		_, err := instancesV2.InstanceMetadata(ctx, node)
```

Comment:
In this case I don't think we even need to call `InstanceMetadata`, can we just return "", nil if the implementation has enabled V2